rustls/src/server/hs.rs

```diff
@@ -362,6 +362,19 @@ impl State for ExpectClientHello {
             ProtocolVersion::TLSv1_2
         };
 
+        if version == ProtocolVersion::TLSv1_3
```

Comment:
Alternatively, we could reject non-`TLSv1_2` legacy versions up front before doing any other processing (or here, but independent of whether we selected 1.3).